server/handlers/common_handlers.go

```diff
@@ -151,3 +152,14 @@ func (h *Handler) DownloadHandler(responseWriter http.ResponseWriter, request *h
 		return
 	}
 }
+
```

Comment:
```suggestion
// Deep-link and redirect support to land user on their originally requested page post authentication instead of dropping user on the root (home) page.
```